builders/txDataBuilder.go

```diff
@@ -0,0 +1,184 @@
+package elrond
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/core"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
+)
+
+const dataSeparator = "@"
+
+// txDataBuilder can be used to easy construct a transaction's data field for a smart contract call
+// can also be used to construct a VmValueRequest instance ready to be used on a VM query
+type txDataBuilder struct {
+	address    string
+	function   string
+	callerAddr string
+	args       []string
+	log        logger.Logger
+	err        error
+}
+
+// NewTxDataBuilder creates a new transaction data builder
+func NewTxDataBuilder(log logger.Logger) *txDataBuilder {
+	builder := &txDataBuilder{
+		log: log,
+	}
+	if check.IfNil(log) {
+		builder.err = ErrNilLogger
+	}
+
+	return builder
+}
+
+// Function sets the function to be called
+func (builder *txDataBuilder) Function(function string) TxDataBuilder {
+	builder.function = function
+
+	return builder
+}
+
+// CallerAddress sets the caller address
+func (builder *txDataBuilder) CallerAddress(address core.AddressHandler) TxDataBuilder {
+	err := builder.checkAddress(address)
+	if err != nil {
+		builder.err = err
+		return builder
+	}
+
+	builder.callerAddr = address.AddressAsBech32String()
+
+	return builder
+}
+
+// Address sets the destination address
+func (builder *txDataBuilder) Address(address core.AddressHandler) TxDataBuilder {
+	err := builder.checkAddress(address)
+	if err != nil {
+		builder.err = err
+		return builder
+	}
+
+	builder.address = address.AddressAsBech32String()
+
+	return builder
+}
+
+// ArgHexString adds the provided hex string to the arguments list
+func (builder *txDataBuilder) ArgHexString(hexed string) TxDataBuilder {
+	_, err := hex.DecodeString(hexed)
+	if err != nil {
+		builder.err = fmt.Errorf("%w in builder.ArgHexString for string %s", err, hexed)
+		return builder
+	}
+
+	builder.args = append(builder.args, hexed)
+
+	return builder
+}
+
+// ArgAddress adds the provided address to the arguments list
+func (builder *txDataBuilder) ArgAddress(address core.AddressHandler) TxDataBuilder {
+	err := builder.checkAddress(address)
+	if err != nil {
+		builder.err = err
+		return builder
+	}
+
+	return builder.addBytes(address.AddressBytes())
+}
+
+func (builder *txDataBuilder) checkAddress(address core.AddressHandler) error {
+	if check.IfNil(address) {
+		return fmt.Errorf("%w in builder.checkAddress", ErrNilAddress)
+	}
+	if len(address.AddressBytes()) == 0 {
```

Comment:
might be checked against the actual number of bytes for a valid address

Reply:
fixed. AddressHandler already have an `IsValid` function. Used that. Thanks